bin/monitor

```diff
@@ -2,62 +2,96 @@
 # frozen_string_literal: true
 
 require_relative "../loader"
+clover_freeze
 
 MONITORABLE_RESOURCE_TYPES = [PostgresServer, MinioServer, GithubRunner]
+resources = {}
+mutex = Mutex.new
 
-sessions = {}
-ssh_threads = {}
-pulse_threads = {}
+producer = Thread.new do
+  loop do
+    mutex.synchronize do
+      MONITORABLE_RESOURCE_TYPES.flat_map { _1.all }.each do |r|
```

Comment:
```suggestion
      Enumerable::Chain.new(*MONITORABLE_RESOURCE_TYPES).each do |r|
```

debatable if it's an improvement, but I thought somehow this was more direct and had fewer materializations.

Reply:
Didn't know Enumarator::Chain. I liked it.